--- a/runtime/Cpp/runtime/CMakeLists.txt
+++ b/runtime/Cpp/runtime/CMakeLists.txt
@@ -157,6 +157,15 @@
 endif()
 
+# Fix macOS linking (before copies are made!)
+if(CMAKE_SYSTEM_NAME MATCHES "Darwin" AND TARGET antlr4_shared)
+  add_custom_command(
+    TARGET antlr4_shared
+    POST_BUILD
+    COMMAND install_name_tool -id "@rpath/$<FILENAME:$<TARGET_FILE:antlr4-runtime>>" $<TARGET_FILE:antlr4-runtime>
+    COMMAND install_name_tool -id "@rpath/$<FILENAME:$<TARGET_LINKER_FILE:antlr4-runtime>>" $<TARGET_LINKER_FILE:antlr4-runtime>)
+endif()
+
 if (ANTLR_BUILD_CPP_TESTS)
   # Copy the generated binaries to dist folder (required by test suite)
   if (TARGET antlr4_shared)
   add_custom_command(
